### PR TITLE
[kf5crash] fix cmake.in to allow consumption by static builds

### PR DESCRIPTION
**Pull request by wrobelda**
- #### What does your PR fix?  
  Includes a patch from https://invent.kde.org/frameworks/kcrash/-/merge_requests/23
  Depends on:
  - #19439 
  - #19440 
- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  No change

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
 Yes

### CODE REVIEW

**Comment by wrobelda** [comment]
@JackBoosY FYI the failures seem unrelated (https://github.com/microsoft/vcpkg/pull/19181)

**Comment by JackBoosY** [comment]
Please note this port only support linux officially. Did you have any questions about that?

**Review by JackBoosY — APPROVED** [review]

**Comment by wrobelda** [comment]
> Please note this port only support linux officially. Did you have any questions about that?

Not sure what you mean by that? It is a Multiplatform API: https://api.kde.org/frameworks-api/frameworks-apidocs/frameworks/kcrash/html/namespaceKCrash.html

File: ports/kf5crash/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "kf5crash",
   "version": "5.84.0",
+  "port-version": 1,
   "description": "KCrash provides support for intercepting and handling application crashes.",
   "homepage": "https://api.kde.org/frameworks/kcrash/html/index.html",
   "supports": "linux",
```
Comment (JackBoosY) [inline ports/kf5crash/vcpkg.json:7]: See here.
In vcpkg, if we declared this, our pipeline test will only test the x64-linux triplet.

Comment (wrobelda) [reply ports/kf5crash/vcpkg.json:7]: Oh wow, didn't notice that. My bad, thanks!

**Comment by wrobelda** [comment]
After enabling compilation on macOS and Windows, this now depends on a fix in kf5windowsystem. Will update here once it's ready.

**Comment by JackBoosY** [comment]
```
[16/18] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1429~1.300\bin\Hostx64\x64\cl.exe   /TP -DKCOREADDONS_LIB -DKF_DEPRECATED_WARNINGS_SINCE=0x060000 -DKF_DISABLE_DEPRECATED_BEFORE_AND_AT=0x055100 -DQT_ACCESSIBILITY_SUPPORT_LIB -DQT_CORE_LIB -DQT_DEPRECATED_WARNINGS_SINCE=0x060000 -DQT_DISABLE_DEPRECATED_BEFORE=0x050f00 -DQT_EVENTDISPATCHER_SUPPORT_LIB -DQT_FONTDATABASE_SUPPORT_LIB -DQT_GUI_LIB -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_CAST_TO_ASCII -DQT_NO_FOREACH -DQT_NO_KEYWORDS -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_NO_URL_CAST_FROM_STRING -DQT_THEME_SUPPORT_LIB -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_WINDOWSUIAUTOMATION_SUPPORT_LIB -DUNICODE -DWIN32_LEAN_AND_MEAN -DWINVER=0x0600 -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -D_UNICODE -D_USE_MATH_DEFINES -D_WIN32_IE=0x0600 -D_WIN32_WINNT=0x0600 -Isrc -ID:\buildtrees\kf5crash\src\v5.84.0-e25e6ced55.clean\src -Isrc\KF5Crash_autogen\include -ID:\installed\x64-windows-static\include\qt5 -ID:\installed\x64-windows-static\include\qt5\QtCore -ID:\installed\x64-windows-static\tools\qt5\mkspecs\win32-msvc -ID:\installed\x64-windows-static\include\KF5\KCoreAddons -ID:\installed\x64-windows-static\include\KF5 -ID:\installed\x64-windows-static\include\KF5\KWindowSystem -ID:\installed\x64-windows-static\include\qt5\QtGui -ID:\installed\x64-windows-static\include\qt5\QtEventDispatcherSupport -ID:\installed\x64-windows-static\include\qt5\QtFontDatabaseSupport -ID:\installed\x64-windows-static\include\qt5\QtThemeSupport -ID:\installed\x64-windows-static\include\qt5\QtAccessibilitySupport -ID:\installed\x64-windows-static\include\qt5\QtWindowsUIAutomationSupport -ID:\installed\x64-windows-static\include\qt5\QtWidgets /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /wd4250 /wd4251 /wd4396 /wd4661 /D_DEBUG /MTd /Z7 /Ob0 /Od /RTC1  -MTd /utf-8 -std:c++17 /showIncludes /Fosrc\CMakeFiles\KF5Crash.dir\kcrash.cpp.obj /Fdsrc\CMakeFiles\KF5Crash.dir\KF5Crash.pdb /FS -c D:\buildtrees\kf5crash\src\v5.84.0-e25e6ced55.clean\src\kcrash.cpp
FAILED: src/CMakeFiles/KF5Crash.dir/kcrash.cpp.obj 
C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1429~1.300\bin\Hostx64\x64\cl.exe   /TP -DKCOREADDONS_LIB -DKF_DEPRECATED_WARNINGS_SINCE=0x060000 -DKF_DISABLE_DEPRECATED_BEFORE_AND_AT=0x055100 -DQT_ACCESSIBILITY_SUPPORT_LIB -DQT_CORE_LIB -DQT_DEPRECATED_WARNINGS_SINCE=0x060000 -DQT_DISABLE_DEPRECATED_BEFORE=0x050f00 -DQT_EVENTDISPATCHER_SUPPORT_LIB -DQT_FONTDATABASE_SUPPORT_LIB -DQT_GUI_LIB -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_CAST_TO_ASCII -DQT_NO_FOREACH -DQT_NO_KEYWORDS -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_NO_URL_CAST_FROM_STRING -DQT_THEME_SUPPORT_LIB -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_WINDOWSUIAUTOMATION_SUPPORT_LIB -DUNICODE -DWIN32_LEAN_AND_MEAN -DWINVER=0x0600 -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -D_UNICODE -D_USE_MATH_DEFINES -D_WIN32_IE=0x0600 -D_WIN32_WINNT=0x0600 -Isrc -ID:\buildtrees\kf5crash\src\v5.84.0-e25e6ced55.clean\src -Isrc\KF5Crash_autogen\include -ID:\installed\x64-windows-static\include\qt5 -ID:\installed\x64-windows-static\include\qt5\QtCore -ID:\installed\x64-windows-static\tools\qt5\mkspecs\win32-msvc -ID:\installed\x64-windows-static\include\KF5\KCoreAddons -ID:\installed\x64-windows-static\include\KF5 -ID:\installed\x64-windows-static\include\KF5\KWindowSystem -ID:\installed\x64-windows-static\include\qt5\QtGui -ID:\installed\x64-windows-static\include\qt5\QtEventDispatcherSupport -ID:\installed\x64-windows-static\include\qt5\QtFontDatabaseSupport -ID:\installed\x64-windows-static\include\qt5\QtThemeSupport -ID:\installed\x64-windows-static\include\qt5\QtAccessibilitySupport -ID:\installed\x64-windows-static\include\qt5\QtWindowsUIAutomationSupport -ID:\installed\x64-windows-static\include\qt5\QtWidgets /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /wd4250 /wd4251 /wd4396 /wd4661 /D_DEBUG /MTd /Z7 /Ob0 /Od /RTC1  -MTd /utf-8 -std:c++17 /showIncludes /Fosrc\CMakeFiles\KF5Crash.dir\kcrash.cpp.obj /Fdsrc\CMakeFiles\KF5Crash.dir\KF5Crash.pdb /FS -c D:\buildtrees\kf5crash\src\v5.84.0-e25e6ced55.clean\src\kcrash.cpp
D:\installed\x64-windows-static\include\KF5\KCoreAddons\kaboutdata.h(30): error C2375: 'KCrash::defaultCrashHandler': redefinition; different linkage
D:\buildtrees\kf5crash\src\v5.84.0-e25e6ced55.clean\src\kcrash.h(56): note: see declaration of 'KCrash::defaultCrashHandler'
```

File: ports/kf5crash/portfile.cmake [hunk 2]
```diff
@@ -4,25 +4,31 @@ vcpkg_from_github(
...
-    PREFER_NINJA
     OPTIONS
         -DBUILD_TESTING=OFF
+        -DBUILD_HTML_DOCS=OFF
+        -DBUILD_MAN_DOCS=OFF
+        -DBUILD_QTHELP_DOCS=OFF
+    MAYBE_UNUSED_VARIABLES
+        BUILD_HTML_DOCS
+        BUILD_MAN_DOCS
+        BUILD_QTHELP_DOCS
```
Comment (JackBoosY) [inline ports/kf5crash/portfile.cmake:22]: Can you confirm these options are exist in the source?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/kf5crash/portfile.cmake (hunk 2 above)
Comment (wrobelda) [reply ports/kf5crash/portfile.cmake:22]: @JackBoosY, as mentioned in another PR, these are standard KDE variables used across KDE/QT software. If at any point a library adds a manual or HTML docs, these will apply. I thought that it makes sense for consistency with other frameworks where such variables are already present in their CMakeLists.txt.

**Comment by wrobelda** [comment]
> D:\installed\x64-windows-static\include\KF5\KCoreAddons\kaboutdata.h(30): error C2375: 'KCrash::defaultCrashHandler': redefinition; different linkage
> D:\buildtrees\kf5crash\src\v5.84.0-e25e6ced55.clean\src\kcrash.h(56): note: see declaration of 'KCrash::defaultCrashHandler'

Yes, saw that, working on it – this PR is back into Draft state, I'll update one I resolve this issue.

File: ports/kf5crash/portfile.cmake (hunk 2 above)
Comment (wrobelda) [reply ports/kf5crash/portfile.cmake]: Removed them based on the discussion here: #19245 (review)

**Comment by wrobelda** [comment]
@JackBoosY all of the dependencies were merged to master. I rebased against it and it's ready for a final review.

File: ports/kf5crash/support_static_builds.patch [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ find_dependency(Qt5Core @REQUIRED_QT_VERSION@)
+ 
++if (NOT @BUILD_SHARED_LIBS@)
++    find_dependency(KF5CoreAddons REQUIRED)
++    find_dependency(KF5WindowSystem REQUIRED)
++
++    if (@X11_FOUND@)
++        find_dependency(Qt5X11Extras REQUIRED)
++    endif()
++endif()
```
Comment (JackBoosY) [inline ports/kf5crash/support_static_builds.patch:25]: This changes didn't match `dependencies` on vcpkg.json, can you explain it?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/kf5crash/support_static_builds.patch (hunk 3 above)
Comment (wrobelda) [reply ports/kf5crash/support_static_builds.patch:25]: It matches them 1:1:
```
"dependencies": [
     "ecm",
     "kf5coreaddons",
     "kf5windowsystem",
     {
       "name": "qt5-x11extras",
       "platform": "linux"
     },
     {
       "name": "vcpkg-cmake",
       "host": true
     },
     {
       "name": "vcpkg-cmake-config",
       "host": true
     }
```

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the fix!

**Pull request merged**